OtripleS.Portal.Web.Tests.Unit/Services/StudentViews/StudentViewServiceTests.Exceptions.RetrieveAll.cs

```diff
@@ -0,0 +1,95 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE AS LONG AS SOFTWARE FUNDS ARE DONATED TO THE POOR
+// ---------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Moq;
+using OtripleS.Portal.Web.Models.StudentViews;
+using OtripleS.Portal.Web.Models.StudentViews.Exceptions;
+using Xunit;
+
+namespace OtripleS.Portal.Web.Tests.Unit.Services.StudentViews
+{
+    public partial class StudentViewServiceTests
+    {
+        [Theory]
+        [MemberData(nameof(DependencyExceptions))]
+        public async Task ShouldThrowStudentViewDependencyExceptionIfDependencyErrorOccursAndLogItAsync(
+           Exception dependencyException)
+        {
+            // given
+            var expectedStudentViewDependencyException =
+                new StudentViewDependencyException(dependencyException);
+
+            this.studentServiceMock.Setup(service =>
+                service.RetrieveAllStudentsAsync())
+                    .ThrowsAsync(dependencyException);
+
+            // when
+            ValueTask<List<StudentView>> retrieveAllStudentsTask =
+                this.studentViewService.RetrieveAllStudentViewsAsync();
+
+            // then
+            await Assert.ThrowsAsync<StudentViewDependencyException>(() =>
+                retrieveAllStudentsTask.AsTask());
+
+            this.studentServiceMock.Verify(service =>
+                service.RetrieveAllStudentsAsync(),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedStudentViewDependencyException))),
+                        Times.Once);
+
+            this.studentServiceMock.VerifyNoOtherCalls();
```

Comment:
We need to have the order specific to this test workflow. So the logging happens before the datetime.

Please verify the order in other tests too.